### PR TITLE
Issue/13328 restore skeleton

### PR DESCRIPTION
**Pull request by zwarm**
Parent #13328 

This PR introduces the "Restore" feature as accessed via the activity log list. This is prep PR.
 
`RestoreActivity`, `RestoreViewModel` are copies of the `BackupDownloadActivity` and `BackupDownloadViewModel` and functionality for observables and the wizard remain the same. Differences include supporting class renames, as well as function name changes. I have left //todo's to fill in particulars for each step as I include them.

`RestoreDetailsFragment` and `RestoreDetailsViewModel` are mere skeletons because it would have been too much to review. 

I did not use the `Rewind` package, but created a new one for `Restore`. I didn't rename because the current classes in `Rewind` are used by the ActivityLog. These `rewind` classes may be removed in the future and if so, this package will be deleted in its entirety.

This PR is a bit larger than normal, but to show that it works I had to go all the way to presenting the details view. It's best reviewed by commits which I have broken down as follows:

- Dagger & App Housekeeping: 9e4ad1b 
- Feature Flag in ActivityLog: ad0cc55
-- Adds support for showing the new restore wizard when the `RestoreFeatureConfig` is enabled. This will work with or without the BackupFeatureConfig flag.
- Restore Activity/Details: 723852c
-- This includes the cloned classes from BackupDownload feature
- Skeleton Unit Tests: 4f9d98b

**Merge Instructions**
- <del>Make sure #13759 has been merged<del> ✅
- <del>Remove Not Ready For Merge Label<del> ✅
- Merge as normal

**To test:**
- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag off
- Navigate to ActivityLog
- Tap the rewind icon for an item
- Note that an empty screen with the "Restore" title is shown

- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag on
- Navigate to ActivityLog
- Tap the More menu
- Tap Restore to this point menu item
- Note that an empty screen with the "Restore" title is shown

- Launch the app with RestoreFeatureConfig flag off and the BackupFeatureFlag off
- Navigate to ActivityLog
- Tap the rewind icon for an item
- Note that the restore dialog box is shown

PR submission checklist:
- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt]: Idea (💡): I wonder if moving the `if/else` logic to the view model would be better? This will help reduce the complexity on the activity level (keep it as dump as possible) and help with testing this logic as well. Wdyt? For example, having `ShowRestoreScreen` and `ShowRestoreDialog` out of `ShowRestore`, which will be then driven by the view model.

PS: Same applies for the `if/else` logic further above on the `showRewindDialog` live data object.
PSS: I know that this logic would be eventually removed at some point, but until them and since this will land on `develop`, maybe it is worth the extra effort.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt:110]: Question (❓): Why not adding the `PROGRESS.id` and `COMPLETE.id` cases to match that with the `BackupDownload` equivalent view model logic? Especially since the `RestoreInProgress` and `RestoreCompleted` objects are already there? Will that logic change for the `Restore` screen?

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt:167]: 1) Question (❓): This is currently unused. What if this is being used by the `RestoreDetailsViewModel`, just like it happens for the `BackupDownload` equivalent code. Will it be used as such in the future?
2) Question (❓): Since the `DetailsToolbarState` is added, I am guessing that more of that would become available later on in the future. If my assumption is correct, why not add all of them now even if the strings/icons are not known yet?

File: WordPress/src/main/res/layout/restore_details_fragment.xml
Comment (ParaskP7) [inline WordPress/src/main/res/layout/restore_details_fragment.xml:6]: 1) Minor (🔍): This layout file can be reformatted.
2) Question (❓): On the `BackupDownload` equivalent layout file, there is an unused `android:id` and an extra `android:padding`. I am guessing we can remove the unused `android:id` from there to increase consistency between the two layout files. However, I am not sure about the `android:padding` and whether this is required for the `Restore` case.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/restore/RestoreDetailsViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/restore/RestoreDetailsViewModelTest.kt:1]: Warning (⚠️): This test resides in the wrong package. Consider moving that to an inner `details` package to match that with the `main` source set structure.

**Review by ParaskP7 — COMMENTED** [review]
👋 @zwarm !

I have review and tested this PR as per the instructions. Everything works as expected, great work here! 🌟 

1) Question (❓): Should there be a third entry point in addition to the two that got updated based on the feature flag. This third entry point would be from within the `ActivityLogDetailsFragment`, see `onRewindButtonClicked(...)`. Should this be changed to navigate to the newly added `Restore` screen as well? That is, instead of showing the `Restore Site` dialog?
2) Info (ℹ️): Most of my comments below was based on my diffing between the existing `BackupDownload` and new `Restore` files, this is how I reviewed this PR. As such, the questions (❓) I raised are based on that fact. Bear that in mind when reviewing them further below.

PS: Let me know how to process and I will approve the PR afterwards, there are no breaking changes, mainly minors that I want to give you a change to fix before merging this PR to `develop`.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/restore/RestoreDetailsViewModelTest.kt
Comment (zwarm) [reply WordPress/src/test/java/org/wordpress/android/ui/jetpack/restore/RestoreDetailsViewModelTest.kt:1]: I didn't put any of the tests into packages. I shall task this separately because it would need to be done for backup download as well. Thanks.

File: WordPress/src/main/res/layout/restore_details_fragment.xml
Comment (zwarm) [reply WordPress/src/main/res/layout/restore_details_fragment.xml:6]: Great comment - I'll be focusing on styling next week. I am aware of the padding already. :)

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt:167]: 1. It will be .. just the prep PR.
2. Trying to keep the pr's code relevant to specific topics. The other toolbar states will be added with each step implementation.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt:110]: They will be added as I move onto those steps - tried to keep this PR focused to just enough so that restore is shown

File: WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt]: > I know that this logic would be eventually removed at some point

Lol, but great idea. Let me implement and push the commit back. 👍

Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt]: @ParaskP7 in commit e081fe2. Can you give it a whirl

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt]: Love it @zwarm ! 🌟 

PS: Since now we have additional flows on the view model, mainly creating a couple of test for those flows would be beneficial, wdyt? Just copy-paste the existing tests and add the differentiator factor. This way we would have full test coverage for this feature.

**Comment by ParaskP7** [comment]
> Question (❓): Should there be a third entry point in addition to the two that got updated based on the feature flag. This third entry point would be from within the ActivityLogDetailsFragment, see onRewindButtonClicked(...). Should this be changed to navigate to the newly added Restore screen as well? That is, instead of showing the Restore Site dialog?

How about this question @zwarm ?

**Comment by zwarm** [comment]
> How about this question @zwarm ?

It's on my task list already as a separate item. :)

**Review by ParaskP7 — APPROVED** [review]
Approving now, thanks for the updates and for making sure that those changes will happen on a subsequent PR @zwarm ! 🌟

**Pull request merged**
